banzai_floyds/orders.py

```diff
@@ -0,0 +1,141 @@
+    order_mask = np.logical_and(order_mask, np.logical_and(x2d >= center.domain[0], x2d <= center.domain[1]))
+    return order_mask
+
+
+def estimate_order_centers(data, error, order_width, peak_separation=10, min_signal_to_noise=100.0):
+    matched_filtered = np.zeros(data.shape[0])
+    for i in np.arange(data.shape[0]):
+        # Run a matched filter using a top hat filter
+        filter_model = Legendre([i], domain=(0, data.shape[1] - 1))
+
+        filter_region = order_region(order_width, filter_model, data.shape)
+        matched_filtered[i] = tophat_filter_metric(data, error, filter_region)
+    peaks = matched_filtered == maximum_filter1d(matched_filtered, size=peak_separation, mode='constant', cval=0.0)
+    peaks = np.logical_and(peaks, matched_filtered > min_signal_to_noise)
+    # Why we have to use flatnonzero here instead of argwhere behaving the way I want is a mystery
+    return np.flatnonzero(peaks)
+
+
+def evaluate_order_model(theta, data, error, order_width):
+    # Set the parameters of the model (polynomial) object
+    model = Legendre(theta, domain=(0, data.shape[1] - 1))
+
+    # Convert the model into a boolean region array
+    order_mask = order_region(order_width, model, data.shape)
+
+    # TODO: Implement a smoothed filter so the edges aren't so sharp. Use two logistic functions for each of the edges
+    # Note the normalization will need a square of the weights / sigma^2. This is due to combining uncertainty
+    # propagation for a weighted sum w^2 sigma^2 (the w = P / sigma^2 so one pair of the sigmas cancel)
+    # Evaluate the metric
+    return tophat_filter_metric(data, error, order_mask)
+
+
+def fit_order_curve(data, error, order_width, initial_guess):
+    # Note that having too high of signal to noise actually makes the gradient less smooth so the gradients will become
+    # discontinuous. In our unit tests, it typically only led to a couple of pixels being off but convergence does
+    # become more difficult.
+    best_fit = minimize(lambda *args: -evaluate_order_model(*args), initial_guess, args=(data, error, order_width),
+                        method='Powell', options={'xtol': 1e-7, 'ftol': 1e-8, 'maxfev': 1e6})
+    return Legendre(best_fit.x, domain=(0, data.shape[1] - 1))
+
+
+class OrderLoader(CalibrationUser):
+    def on_missing_master_calibration(self, image):
+        if image.obstype == 'SKYFLAT':
+            return image
+        else:
+            return super(OrderLoader, self).on_missing_master_calibration(image)
+
+    @property
+    def calibration_type(self):
+        return 'SKYFLAT'
+
+    def apply_master_calibration(self, image, master_calibration_image):
+        image.orders = master_calibration_image.orders
+        return image
+
+
+class OrderSolver(Stage):
+    # Currently we hard code the order width to 93. If we wanted to measure it I recommend using a Canny filter and
+    # taking the edge closest to the previous guess of the edge.
+    ORDER_WIDTH = 93
```

Comment:
should this be `ORDER_HEIGHT`?
might be a minor difference, but I think it might avoid some confusion.